src/convertToObject.js

```diff
@@ -6,7 +6,19 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const sourceStringParts = sourceString.split(';');
+  const strWithoutSpaces = sourceStringParts.map((str) => str.trim());
+  const result = strWithoutSpaces.reduce((prev, el) => {
+    const [key, value] = el.split(':');
+
+    if (key && value) {
+      prev[key.trim()] = value.trim();
+    }
+
+    return prev;
```

Comment:
[CODE PRACTICE] - Instead of using the `reduce` method inside the `convertToObject` function, consider using `filter` to remove any empty strings after splitting by semicolon and then map to create the final object. This aligns with the checklist item to avoid loops and use iteration methods instead.